lib/sentry/opentelemetry/span_record.ex

```diff
@@ -0,0 +1,72 @@
+if Code.ensure_loaded?(OpenTelemetry) do
+  defmodule Sentry.OpenTelemetry.SpanRecord do
+    require Record
+    require OpenTelemetry
+
+    @fields Record.extract(:span, from_lib: "opentelemetry/include/otel_span.hrl")
+    Record.defrecordp(:span, @fields)
```

Comment:
Easier to read code with this, IMO?

```suggestion
    Record.defrecordp(:otel_span, @fields)
```

Reply:
This breaks:

```
     ** (ArgumentError) expected argument to be a literal atom, literal keyword or a :otel_span record, got runtime: {:span, 135612037030898638558808201682891567440, 9466375428462860189, [], :undefined, "instrumented_function", :internal, -576460748421781665, :undefined, {:attributes, 128, :infinity, 0, %{}}, {:events, 128, 128, :infinity, 0, []}, {:links, 128, 128, :infinity, 0, []}, :undefined, 1, true, :undefined}
```

Reply:
This works just fine for me:

```
diff --git lib/sentry/opentelemetry/span_record.ex lib/sentry/opentelemetry/span_record.ex
index 0bfdae3..361f961 100644
--- lib/sentry/opentelemetry/span_record.ex
+++ lib/sentry/opentelemetry/span_record.ex
@@ -4,12 +4,12 @@ if Code.ensure_loaded?(OpenTelemetry) do
     require OpenTelemetry
 
     @fields Record.extract(:span, from_lib: "opentelemetry/include/otel_span.hrl")
-    Record.defrecordp(:span, @fields)
+    Record.defrecordp(:otel_span, @fields)
 
     defstruct @fields ++ [:origin]
 
     def new(otel_span) do
-      otel_attrs = span(otel_span)
+      otel_attrs = otel_span(otel_span)
 
       {:attributes, _, _, _, attributes} = otel_attrs[:attributes]
 ```

Reply:
I don't think this conversation is resolved?

Reply:
@whatyouhide it does not work:

```
  1) test removes span records from storage after sending a transaction (Sentry.Opentelemetry.SpanProcessorTest)
     test/sentry/opentelemetry/span_processor_test.exs:86
     ** (ArgumentError) expected argument to be a literal atom, literal keyword or a :otel_span record, got runtime: {:span, 33858799209017318098040011039887091530, 12090113571782940245, [], :undefined, "instrumented_function", :internal, -576460750127872124, :undefined, {:attributes, 128, :infinity, 0, %{}}, {:events, 128, 128, :infinity, 0, []}, {:links, 128, 128, :infinity, 0, []}, :undefined, 1, true, :undefined}
     code: TestEndpoint.instrumented_function()
     stacktrace:
       (elixir 1.18.1) lib/record.ex:516: Record.__keyword__/3
       (sentry 10.9.0) lib/sentry/opentelemetry/span_record.ex:14: Sentry.OpenTelemetry.SpanRecord.new/1
       (sentry 10.9.0) lib/sentry/opentelemetry/span_processor.ex:20: Sentry.OpenTelemetry.SpanProcessor.on_start/3
       (stdlib 6.1.2) lists.erl:2146: :lists.foldl/3
       (opentelemetry 1.5.0) /usr/local/src/sentry-elixir/deps/opentelemetry/src/otel_span_ets.erl:64: :otel_span_ets.start_span/7
       (opentelemetry 1.5.0) /usr/local/src/sentry-elixir/deps/opentelemetry/src/otel_tracer_default.erl:37: :otel_tracer_default.start_span/4
       (opentelemetry 1.5.0) /usr/local/src/sentry-elixir/deps/opentelemetry/src/otel_tracer_default.erl:43: :otel_tracer_default.with_span/5
       test/sentry/opentelemetry/span_processor_test.exs:91: (test)
```

Reply:
Can you push up a branch where this fails? I’m suspecting there's somewhere else in the codebase where you might be calling `span(...)` in a way that passes `{:span, ...}` where it's supposed to be `:otel_span`.

Reply:
@whatyouhide this comes from otel tracer that we don't control, it passes `{:span, ....}` to `on_start` so it crashes when the tuple is defined as `{:otel_span, ...}`.

Reply:
Here's exception from one of the tests:

```
  1) test sends captured root spans as transactions (Sentry.Opentelemetry.SpanProcessorTest)
     test/sentry/opentelemetry/span_processor_test.exs:28
     ** (FunctionClauseError) no function clause matching in Sentry.OpenTelemetry.SpanRecord.new/1

     The following arguments were given to Sentry.OpenTelemetry.SpanRecord.new/1:

         # 1
         {:span, 6348120493686974701910169917510003045, 12343669577700150675, [], :undefined, "child_instrumented_function_one", :internal, -576460750972794583, :undefined, {:attributes, 128, :infinity, 0, %{}}, {:events, 128, 128, :infinity, 0, []}, {:links, 128, 128, :infinity, 0, []}, :undefined, 1, true, :undefined}

     Attempted function clauses (showing 1 out of 1):

         def new({:otel_span, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _} = otel_span)

     code: TestEndpoint.child_instrumented_function("one")
     stacktrace:
       (sentry 10.9.0) lib/sentry/opentelemetry/span_record.ex:13: Sentry.OpenTelemetry.SpanRecord.new/1
       (sentry 10.9.0) lib/sentry/opentelemetry/span_processor.ex:20: Sentry.OpenTelemetry.SpanProcessor.on_start/3
       (stdlib 6.1.2) lists.erl:2146: :lists.foldl/3
       (opentelemetry 1.5.0) /usr/local/src/sentry-elixir/deps/opentelemetry/src/otel_span_ets.erl:64: :otel_span_ets.start_span/7
       (opentelemetry 1.5.0) /usr/local/src/sentry-elixir/deps/opentelemetry/src/otel_tracer_default.erl:37: :otel_tracer_default.start_span/4
       (opentelemetry 1.5.0) /usr/local/src/sentry-elixir/deps/opentelemetry/src/otel_tracer_default.erl:43: :otel_tracer_default.with_span/5
       test/sentry/opentelemetry/span_processor_test.exs:33: (test)
```

Reply:
> Can you push up a branch where this fails? I’m suspecting there's somewhere else in the codebase where you might be calling `span(...)` in a way that passes `{:span, ...}` where it's supposed to be `:otel_span`.

I pushed a branch called 874-add-spanprocessor-for-otel-record-rename with this commit  177415a5691c2d851c8fd22a3cca233925fafbdf